### PR TITLE
Dedup vcardstore images, only request vcards for ONLINE users

### PR DESCRIPTION
**Pull request by ryscheng**
Basically issues
https://github.com/freedomjs/freedom-social-xmpp/issues/91
https://github.com/freedomjs/freedom-social-xmpp/issues/95

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -212,7 +212,8 @@ module.exports = function(grunt) {
     'build',
     'copy:jasmine',
     'jasmine',
-    'jasmine_node'
+    'jasmine_node',
+    //'jasmine_chromeapp'
```
Comment (dborkan) [inline Gruntfile.js]: you can probably revert this

File: src/core/socialprovider.js [hunk 2]
```diff
@@ -437,14 +436,14 @@ XMPPSocialProvider.prototype.updateRoster = function(msg) {
...
-  // Response to Query
+  // Response to roster query
   if (query && query.attrs.xmlns === 'jabber:iq:roster') {
     items = query.getChildren('item');
     for (i = 0; i < items.length; i += 1) {
       if(items[i].attrs.jid && items[i].attrs.name) {
         this.vCardStore.updateUser(items[i].attrs.jid, 'name',
             items[i].attrs.name);
-        this.vCardStore.refreshContact(items[i].attrs.jid);
+        //this.vCardStore.refreshContact(items[i].attrs.jid);
```
Comment (dborkan) [inline src/core/socialprovider.js]: want to just remove this, or comment on it?

**Comment by dborkan** [comment]
A few small comments to be removed, otherwise :+1: - looking forward to seeing this in action!

**Comment by iislucas** [comment]
Consider adding a test to avoid regression/other bad storage behaviour in future?

**Pull request merged**